akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -458,6 +498,19 @@ class ShardRegion(
     }
   }
 
+  def receiveRegionStateQuery(ref: ActorRef): Unit = {
+    import akka.pattern.{ ask, pipe }
+    import scala.concurrent.duration._
+    implicit val timeout: Timeout = 1.seconds
+    Future.sequence(shards.map {
+      case (shardId, ref) ⇒
+        (ref ? Shard.GetCurrentShardState).mapTo[Shard.CurrentShardState]
+    }).map { shardStates ⇒
+      CurrentShardRegionState(shardStates.map(state ⇒
+        ShardRegion.ShardState(state.shardId, state.entityIds)).toSet)
+    }.pipeTo(ref)
```

Comment:
how should we handle timeout? sending back Status.Failure is too general in my opinion, perhaps we need another reply message for the the error case?